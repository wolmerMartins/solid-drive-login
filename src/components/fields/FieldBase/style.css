.FieldBase {
  border: none;
  border-bottom: 1px solid #CCC;
}

.FieldBase-label {
  font-size: var(--s-font);
}
