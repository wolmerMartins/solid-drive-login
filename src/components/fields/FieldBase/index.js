import React from 'react'

import './style.css'

const FieldBase = ({
  id,
  label,
  children
}) => (
  <fieldset className="FieldBase">
    <label
      htmlFor={id}
      className="FieldBase-label"
    >
      {label}
    </label>
    {children}
  </fieldset>
)

export default FieldBase
