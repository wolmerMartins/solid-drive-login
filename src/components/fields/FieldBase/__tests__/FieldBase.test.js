import React, { useState } from 'react'
import { render, fireEvent } from '@testing-library/react'

import FieldBase from '../index'

const TestComponent = props => {
  const [value, setValue] = useState('')

  const onChange = e => setValue(e.target.value)

  return (
    <FieldBase {...props}>
      <input
        value={value}
        onChange={onChange}
        {...props}
      />
    </FieldBase>
  )
}

const buildComponent = props => {
  const utils = render(<TestComponent {...props} />)
  const input = utils.getByLabelText(props.label)

  return {
    input,
    ...utils
  }
}

describe('<FieldBase />', () => {
  it('Render the component with \'Usuário\' label text', () => {
    const { asFragment } = buildComponent({ id: 'user', label: 'Usuário' })

    expect(asFragment())
      .toMatchSnapshot()
  })

  it('Find the input by label and types a value in', () => {
    const { input } = buildComponent({ id: 'user', label: 'Usuário' })

    fireEvent.change(input, { target: { value: 'test' } })

    expect(input)
      .toHaveProperty('value', 'test')
  })
})
