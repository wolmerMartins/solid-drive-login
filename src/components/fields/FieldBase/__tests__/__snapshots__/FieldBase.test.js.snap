// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldBase /> Render the component with 'Usuário' label text 1`] = `
<DocumentFragment>
  <fieldset
    class="FieldBase"
  >
    <label
      class="FieldBase-label"
      for="user"
    >
      Usuário
    </label>
    <input
      id="user"
      label="Usuário"
      value=""
    />
  </fieldset>
</DocumentFragment>
`;
